library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -119,45 +187,58 @@ public T getItem(int position) {
     return items.get(position);
   }
 
+  /**
+   * Get footer data in this adapter, you should previously use {@link #setFooter(F footer)}
+   * in the adapter initialization code to set footer data.
+   *
+   * @return footer data
+   */
   public F getFooter() {
     return footer;
   }
 
+  /**
+   * If you need a header, you should set header data in the adapter initialization code.
+   *
+   * @param header header data
+   */
   public void setHeader(H header) {
     this.header = header;
   }
 
+  /**
+   * You should set header data in the adapter initialization code.
+   *
+   * @param items item data list
+   */
   public void setItems(List<T> items) {
     validateItems(items);
     this.items = items;
   }
 
+  /**
+   * If you need a footer, you should set footer data in the adapter initialization code.
+   */
   public void setFooter(F footer) {
     this.footer = footer;
   }
 
+  /**
+   * Call this method to show hiding footer.
+   */
   public void showFooter() {
     this.showFooter = true;
     notifyDataSetChanged();
   }
 
+  /**
+   * Call this method to hide footer.
+   */
   public void hideFooter() {
     this.showFooter = false;
     notifyDataSetChanged();
   }
 
-  protected abstract VH onCreateHeaderViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract VH onCreateItemViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract VH onCreateFooterViewHolder(ViewGroup parent, int viewType);
-
-  protected abstract void onBindHeaderViewHolder(VH holder, int position);
-
-  protected abstract void onBindItemViewHolder(VH holder, int position);
-
-  protected abstract void onBindFooterViewHolder(VH holder, int position);
-
```

Comment:
Just moved below `onCreateViewHolder(ViewGroup parent, int viewType)` and `onBindViewHolder(VH holder, int position)`.